packages/calypso-e2e/src/lib/pages/site-import-page.ts

```diff
@@ -0,0 +1,73 @@
+import { Page } from 'playwright';
+
+export type SourceService = typeof SiteImportPage.services[ number ];
+
+const selectors = {
+	// Listing
+	service: ( name: SourceService ) => `h1:text("${ name }")`,
+
+	// Importer screen buttons and pseudo-buttons
+	backButton: 'a:text("Back")',
+	cancelButton: 'button:text("Cancel")',
+	continueButton: 'button:text("Continue")',
+
+	// Inputs
+	siteInput: '[placeholder="example.com"]',
+	fileInput: 'input[name="exportFile"]',
+	fileInputText: 'text=Drag a file here, or click to upload a file',
+};
+
+/**
+ * Class representing the Site Import page.
+ */
+export class SiteImportPage {
+	static services = [ 'WordPress', 'Blogger', 'Medium', 'Squarespace', 'Wix' ] as const;
+	private page: Page;
+
+	/**
+	 * Constructs an instance.
+	 *
+	 * @param {Page} page Underlying page on which interactions take place.
+	 */
+	constructor( page: Page ) {
+		this.page = page;
+	}
+
+	/**
+	 * Select the target service from which site import should take place.
+	 *
+	 * @param {SourceService} service Allowed list of services.
+	 */
+	async selectService( service: SourceService ): Promise< void > {
+		await this.page.click( selectors.service( service ) );
+		await this.page.waitForLoadState( 'load' );
```

Comment:
Should these two be wrapped with a `Promise.all`?

I can see we're also waiting for the `load` event [in the `cancel()` method](https://github.com/Automattic/wp-calypso/pull/55260/files#diff-a7a8195b6397d4f786c496360f2ea159e7489876b3703cf4f72f1c1fe5842296R64), only there we're calling it kinda before the action. Can we assume that when calling `cancel()` the page is already loaded or that _wait for page loaded_ is necessary there as well?

Reply:
Yeah, that is a good assumption. I've removed the unnecessary `waitForLoadState` from Line 77, and grouped the two calls at Line 42-43 into a `Promise.all`.